src/library/scala/collection/immutable/StrictOptimizedSeqOps.scala

```diff
@@ -38,7 +38,7 @@ trait StrictOptimizedSeqOps[+A, +CC[_], +C]
       b += it.next()
       i += 1
     }
-    if (!it.hasNext) throw new IndexOutOfBoundsException(index.toString)
+    if (!it.hasNext) throw new IndexOutOfBoundsException(if (knownSize >= 0) s"can't update at $index since bound is 0-${knownSize - 1}" else index.toString)
```

Comment:
Likewise, we know here that index is larger than the max index, which is i

Reply:
Followed this suggestion.
After iteration, we know the max index, even if not-knownSize Seq